evcouplings/compare/protocol.py

```diff
@@ -221,6 +226,168 @@ def _discrete_count(x):
     return cm_files
 
 
+def _make_complex_contact_maps(ec_table, d_intra_i, d_multimer_i,
+                               d_intra_j, d_multimer_j,
+                               d_inter, first_segment_name,
+                               second_segment_name, **kwargs):
+    """
+       Plot contact maps with all ECs above a certain probability threshold,
+       or a given count of ECs
+
+        Parameters
+        ----------
+        ec_table : pandas.DataFrame
+            Full set of evolutionary couplings (all pairs)
+        d_intra_i, d_intra_j: DistanceMap
+            Computed residue-residue distances within chains for
+            monomers i and j
+        d_multimer_i, d_multimer_j : DistanceMap
+            Computed residue-residue distances between homomultimeric
+            chains for monomers i and j
+        d_inter: DistanceMap
+            Computed residue-residue distances between heteromultimeric
+            chains
+        first_segment_name, second_segment_name: str
+            Name of segment i and segment j in the ec_table
+        **kwargs
+            Further plotting parameters, see check_required in code
+            for necessary values.
+
+        Returns
+        -------
+        cm_files : list(str)
+            Paths of generated contact map files
+        """
+
+    def plot_complex_cm(ecs_i, ecs_j, ecs_inter, 
+                        first_segment_name,
+                        second_segment_name, output_file=None):
+        """
+        Simple wrapper for contact map plotting
+        """
+        with misc.plot_context("Arial"):
+
+            if kwargs["scale_sizes"]:
+                # to scale sizes, combine all ecs to rescale together
+                ecs = pd.concat([ecs_i, ecs_j, ecs_inter])
+                ecs.loc[:, "size"] = ecs.cn.values / ecs.cn.max()
+
+                # split back into three separate DataFrames
+                ecs_i = ecs.query("segment_i == segment_j == @first_segment_name")
+                ecs_j = ecs.query("segment_i == segment_j == @second_segment_name")
+                ecs_inter = ecs.query("segment_i != segment_j")
+
+                # if any of these groups are entry, replace with None
+                if len(ecs_i) == 0:
+                    ecs_i = None
+                if len(ecs_j) == 0:
+                    ecs_j = None
+                if len(ecs_inter) == 0:
+                    ecs_inter = None
+                
+            # Currently, we require at least one of the monomer to have some contacts to make the plot
+            if ecs_i is None and ecs_j is None:
```

Comment:
Actually I am not sure how sensible this requirement is.. what if one wants to plot just experimental structure contacts? The only requirement should be that there is something for either monomer (could be experiment or could be ECs)